settings/cache.settings.php

```diff
@@ -19,9 +20,107 @@
  * are cached permanently and may result in stale configuration when
  * rebuilding the service container on deploymemnts and cache rebuilds.
  * Uncomment the relevant lines for each bin below and configure the desired
- * See https://www.drupal.org/node/2754947.
+ *
+ * @see https://www.drupal.org/node/2754947
  */
 
 // $settings['cache']['bins']['bootstrap'] = 'cache.backend.null';
 // $settings['cache']['bins']['discovery'] = 'cache.backend.null';
 // $settings['cache']['bins']['config'] = 'cache.backend.null';.
+/**
+ * Use memcache as cache backend.
+ *
+ * Autoload memcache classes and service container in case module is not
+ * installed. Avoids the need to patch core and allows for overriding the
+ * default backend when installing Drupal.
+ *
+ * @see https://www.drupal.org/node/2766509
+ */
+if ($is_ah_env) {
+  switch ($ah_env) {
+    case 'test':
+    case 'prod':
+
+      // Check for PHP Memcached libraries.
+      $memcache_exists = class_exists('Memcache', FALSE);
+      $memcached_exists = class_exists('Memcached', FALSE);
+      $memcache_module_is_present = file_exists(DRUPAL_ROOT . '/modules/contrib/memcache/memcache.services.yml');
```

Comment:
The DrupalVM example should be updated to match this logic.

This only real difference is that on DrupalVM you need this:

```
// Include a unique prefix for each local Drupal installation.
if ($is_local_env) {
  $settings['memcache']['key_prefix'] = $site_dir;
}
```